external control plane doc updates (#8471)
istio/istio.io: content/en/docs/setup/additional-setup/external-controlplane/index.md

---

> title: Install External Control Plane Istio

Reviewer: I think we should stick with the original title since `External Control Plane Istio` isn't really a thing.

external control plane [deployment model](/docs/ops/deployment/deployment-models/#control-plane-models)
enables mesh operators to install and manage mesh control planes on separate
external clusters. This deployment model allows a clear separation between mesh
operators and mesh admins. The mesh operators can install and manage the Istio control planes

> and the mesh admins control the mesh resources.

Reviewer: ```suggestion
while the mesh admins only need to configure the mesh resources.
```

## Environment Variables

The following environment variables will be used throughout to simplify the instructions:

Variable | Description
-------- | -----------
`CTX_EXTERNAL_CLUSTER` | The context name in the default [Kubernetes configuration file](https://kubernetes.io/docs/tasks/access-application-cluster/configure-access-multiple-clusters/) used for accessing the external control plane cluster.
`CTX_REMOTE_CLUSTER` | The context name in the default [Kubernetes configuration file](https://kubernetes.io/docs/tasks/access-application-cluster/configure-access-multiple-clusters/) used for accessing the remote cluster.
`EXTERNAL_ISTIOD_ADDR` | The hostname for the ingress gateway on the external control plane cluster. This is used by the `remote_cluster` to access the external control plane.
`SSL_SECRET_NAME` | The name of the secret that holds the TLS certs for the ingress gateway on the external control plane cluster.

Set the `CTX_EXTERNAL_CLUSTER` and `CTX_REMOTE_CLUSTER` now. You will set the others later.

{{< text bash >}}
$ export CTX_EXTERNAL_CLUSTER=external_cluster
$ export CTX_REMOTE_CLUSTER=remote_cluster
{{< /text >}}

## Cluster configuration

> ### Set up the ingress in the external cluster

Reviewer: ```suggestion
### Set up a gateway in the external cluster
```

EOF
{{< /text >}}

Install the configuration to create the ingress gateway in the `istio-system` namespace of `external_cluster`:

{{< text bash >}}

> $ istioctl install -f controlplane-gateway.yaml --context="${CTX_EXTERNAL_CLUSTER}"

Reviewer 2: Another good catch!  This made me want to add istioctl apply cmd to alias to istioctl install :)

EOF
{{< /text >}}

Install the configuration to create the ingress gateway in the `istio-system` namespace of `external_cluster`:

{{< text bash >}}
$ istioctl install -f controlplane-gateway.yaml --context="${CTX_EXTERNAL_CLUSTER}"
{{< /text >}}

> You may notice the istiod deployment created in the istio-system namespace. This is used only to configure the ingress gateway and is NOT the control plane used by remote clusters.

Reviewer: Maybe add another sentence about the actual external istiod, something like this?
```suggestion
You may notice the istiod deployment created in the istio-system namespace. This is used only to configure the ingress gateway and is NOT the control plane used by remote clusters.
This ingress gateway could, in fact, be configured to host multiple external control control planes, in different namespaces on the cluster, even though in this example you will only deploy a single external istiod in the `external-istiod` namespace.
```

> Configure your environment to expose the Istio ingress gateway service as a hostname with TLS. Set the `EXTERNAL_ISTIOD_ADDR` environment variable to the hostname and `SSL_SECRET_NAME` environment variable to the secret that holds the TLS certs:

Reviewer: ```suggestion
Configure your environment to expose the Istio ingress gateway service using a public hostname with TLS. Set the `EXTERNAL_ISTIOD_ADDR` environment variable to the hostname and `SSL_SECRET_NAME` environment variable to the secret that holds the TLS certs:
```

> You may notice the istiod deployment created in the istio-system namespace. This is used only to configure the ingress gateway and is NOT the control plane used by remote clusters. This ingress gateway could, in fact, be configured to host multiple external control control planes, in different namespaces on the cluster, even though in this example you will only deploy a single external istiod in the `external-istiod` namespace.

Reviewer: ```suggestion
You may notice the istiod deployment created in the `istio-system` namespace. This is used only to configure the ingress gateway and is NOT the control plane used by remote clusters. This ingress gateway could, in fact, be configured to host multiple external control control planes, in different namespaces on the cluster, even though in this example you will only deploy a single external istiod in the `external-istiod` namespace.
```

        mode: SIMPLE
        credentialName: $SSL_SECRET_NAME
      hosts:

>       - $EXTERNAL_ISTIOD_ADDR

Reviewer 2: Thanks for catching this, my bad about forgetting the renames.

Apply the configuration on `external_cluster`:

{{< text bash >}}

> $ kubectl create namespace external-istiod --context="${CTX_EXTERNAL_CLUSTER}

Reviewer: ```suggestion
$ kubectl create namespace external-istiod --context="${CTX_EXTERNAL_CLUSTER}"
```

Author: Actually, we don't need this command...I was missing this namespace due to a bug in a different step

  values:
    global:
      caAddress: $EXTERNAL_ISTIOD_ADDR:15012
      istioNamespace: external-istiod
      meshID: mesh1
      multiCluster:
        clusterName: remote_cluster
    istiodRemote:
      injectionURL: https://$EXTERNAL_ISTIOD_ADDR:15017/inject
    base:
      validationURL: https://$EXTERNAL_ISTIOD_ADDR:15017/validate
EOF
{{< /text >}}

Install the configuration on the remote cluster:

{{< text bash >}}
$ istioctl manifest generate -f remote-config-cluster.yaml  | kubectl apply --context="${CTX_REMOTE_CLUSTER}" -f -
{{< /text >}}

> **NOTE:** The ingress gateway will not start until you install the external control plane in the next section.

Reviewer: ```suggestion
**NOTE:** Connections to the ingress gateway in the external cluster will not start working until you install the external control plane in the next section.
```
